p2p/server/server.go

```diff
@@ -362,18 +374,23 @@ func (s *Server) StreamRequest(
 	}
 
 	var srvError *ServerError
-	took := time.Since(start).Seconds()
+	isServerError := errors.As(err, &srvError)
+	took := time.Since(start)
+	tookSecs := took.Seconds()
```

Comment:
I think the naming can be improved:

```suggestion
	duration := time.Since(start).Seconds()
```

wdyt?

Reply:
Actually, `took` (value of type `Duration`) is used as an argument for `info.ClientStats.RequestDone`, whereas `tookSecs` (value of type `float64`) is used for the metrics. Thus, both variables are needed, and renaming `tookSecs` to duration would make things more confusing

Reply:
But it could also be

```go
duration := time.Since(start)

// then use
info.ClientStats.RequestDone(duration, err == nil) // for a `time.Duration` value

// and
s.metrics.clientLatency.Observe(duration.Seconds()) // for the `float64` value
```

or not?

Reply:
Done